sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,223 @@
+|                      | params: shared pointer to parent node, planner name, tf buffer pointer and |                         |
+|                      | shared pointer to costmap.                                                 |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| activate()           | Method is called when planner server enters on_activate state. Ideally this| Yes                     |
+|                      | method should implement operations which are neccessary before planner goes|                         |
+|                      | to an active state.                                                        |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| deactivate()         | Method is called when planner server enters on_deactivate state. Ideally   | Yes                     |
+|                      | this method should implement operations which are neccessary before planner|                         |
+|                      | goes to an inactive state.                                                 |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| cleanup()            | Method is called when planner server goes to on_cleanup state. Ideally this| Yes                     |
+|                      | method should clean up resoures which are created for the planner.         |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| createPlan()         | Method is called when planner server demands a global plan for specified   | Yes                     |
+|                      | start and goal pose. This method returns `nav_msgs::msg::Path` carrying    |                         |
+|                      | global plan. This method takes 2 input parmas: start pose and goal pose.   |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+
+For this tutorial, we will be using methods ``StraightLine::configure()`` and ``StraightLine::createPlan()`` to create straight line planner.
+
+In planners, ``configure()`` method must set member variables from ROS parameters and any initialization required,
+
+.. code-block:: c++
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".interpolation_resolution", rclcpp::ParameterValue(0.1));
```

Comment:
Oh! Oh! One more thing, please example the `name.<param>` thing. That is going to trip a bunch of people up. Explaining that the plugin is mapped to the plugin ID so you can have multiple plugins loaded of the same type. So the `name_.` is used for getting the parameters namespaced to the right version.

Reply:
okay cool!